ambry-clustermap/src/main/java/com/github/ambry/clustermap/ClusterMapUtils.java

```diff
@@ -715,7 +715,19 @@ static class PartitionSelectionHelper implements ClusterMapChangeListener {
       this.clusterManagerQueryHelper = clusterManagerQueryHelper;
       this.defaultPartitionClass = defaultPartitionClass;
       this.clusterManagerMetrics = clusterManagerMetrics;
-      updatePartitions(clusterManagerQueryHelper.getPartitions(), localDatacenterName);
+
+      Collection<? extends PartitionId> partitions = clusterManagerQueryHelper.getPartitions();
+      Collection<PartitionId> filteredPartions = new ArrayList<>();
+      if (clusterManagerQueryHelper.isPartitionFilteringEnabled()) {
+        for (Object partition : partitions) {
```

Comment:
Let's not use Object here, let's change it to PartitionId partition: partitions.